src/Features/Core/Portable/CodeFixes/PreferFrameworkType/PreferFrameworkTypeCodeFixProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.PreferFrameworkType
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic,
+        Name = PredefinedCodeFixProviderNames.PreferFrameworkType), Shared]
+    internal class PreferFrameworkTypeCodeFixProvider : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+                IDEDiagnosticIds.PreferFrameworkTypeInDeclarationsDiagnosticId,
+                IDEDiagnosticIds.PreferFrameworkTypeInMemberAccessDiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider() => BatchFixAllProvider.Instance;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Yup. The idea is to gather up bits here that you need to make any checks that would determine whether you'll register a CodeAction. Since you don't have any checks, you can safely defer this extra work to the CodeAction itself.